--- hidapi.cygport.orig	2015-02-06 11:38:59.000000000 +0900
+++ hidapi.cygport	2021-10-28 07:59:59.777892400 +0900
@@ -1,6 +1,6 @@
 NAME="hidapi"
 VERSION=0.8.0-rc1
-RELEASE=1
+RELEASE=1_1
 CATEGORY="Libs"
 SUMMARY="USB HID library"
 DESCRIPTION="C Library for USB/Bluetooth HID device access from Linux, Mac OS X, FreeBSD, and Windows."
@@ -14,6 +14,9 @@
 https://github.com/signal11/hidapi/commit/54eb31dc16dcc67d0b689ed947bc53a038608c0e.patch
 "
 
-PKG_NAMES="libhidapi0 libhidapi-devel"
-libhidapi0_CONTENTS="usr/bin/cyghidapi-0.dll usr/share/doc/"
-libhidapi_devel_CONTENTS="usr/include/ usr/lib/libhidapi.dll.a usr/lib/pkgconfig/"
+PKG_NAMES="hidapi0 libhidapi-devel"
+hidapi0_CONTENTS="usr/bin/cyghidapi-hidraw-0.dll usr/bin/cyghidapi-libusb-0.dll usr/bin/cyghidapi-0.dll usr/share/doc/"
+libhidapi_devel_CONTENTS="usr/include/ usr/lib/libhidapi-hidraw.dll.a usr/lib/libhidapi-libusb.dll.a usr/lib/libhidapi.dll.a usr/lib/pkgconfig/"
+
+hidapi0_REQUIRES="libusb1.0 libiconv"
+libhidapi_devel_REQUIRES="libusb1.0-devel libiconv-devel"
