--- hidapi.cygport.orig	2015-02-06 11:38:59.000000000 +0900
+++ hidapi.cygport	2021-10-27 13:08:33.971236800 +0900
@@ -12,8 +12,12 @@
 https://github.com/signal11/hidapi/commit/627f3aa478e8099e3660d6a429f1584eeca0724c.patch
 https://github.com/signal11/hidapi/commit/171a521b91add62c19343100df62b15ecd8f3216.patch
 https://github.com/signal11/hidapi/commit/54eb31dc16dcc67d0b689ed947bc53a038608c0e.patch
+https://raw.githubusercontent.com/fmrns/hidapi-cygwin-libusb/5f7c7651b4bbc8b561ca7d584f85016fb8e4207b/hidapi-0.8.0-rc1-1.src.patch
 "
 
-PKG_NAMES="libhidapi0 libhidapi-devel"
-libhidapi0_CONTENTS="usr/bin/cyghidapi-0.dll usr/share/doc/"
-libhidapi_devel_CONTENTS="usr/include/ usr/lib/libhidapi.dll.a usr/lib/pkgconfig/"
+PKG_NAMES="hidapi0 libhidapi-devel"
+hidapi0_CONTENTS="usr/bin/cyghidapi-hidraw-0.dll usr/bin/cyghidapi-libusb-0.dll usr/bin/cyghidapi-0.dll usr/share/doc/"
+libhidapi_devel_CONTENTS="usr/include/ usr/lib/libhidapi-hidraw.dll.a usr/lib/libhidapi-libusb.dll.a usr/lib/libhidapi.dll.a usr/lib/pkgconfig/"
+
+hidapi0_REQUIRES="libusb1.0 libiconv2"
+libhidapi_devel_REQUIRES="libusb1.0-devel libiconv-devel"
